File: src/main/java/net/softbell/bsh/domain/entity/MemberLoginLog.java
package net.softbell.bsh.domain.entity;

import java.util.Date;

import javax.persistence.Column;
import javax.persistence.EmbeddedId;
import javax.persistence.Entity;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;


/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 회원 로그인 로그 엔티티
 */
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@Entity
@Table(name="member_login_log")
@NamedQuery(name="MemberLoginLog.findAll", query="SELECT m FROM MemberLoginLog m")
public class MemberLoginLog
{
	@EmbeddedId
	private MemberLoginLogPK id;

	@Column(nullable=false, length=15)
	private String ipv4;

	@Temporal(TemporalType.TIMESTAMP)
	@Column(name="request_date", nullable=false)
	private Date requestDate;

	@Column(nullable=false)
	private byte status;

	@ManyToOne
	@JoinColumn(name="member_id", nullable=false, insertable=false, updatable=false)
	private Member member;
}

File: src/main/kotlin/net/softbell/bsh/domain/entity/MemberLoginLog.kt
package net.softbell.bsh.domain.entity

import net.softbell.bsh.domain.AuthStatusRule
import java.io.Serializable
import java.util.*
import javax.persistence.*

/**
 * @Author : Bell(bell@softbell.net)
 * @Description : 회원 로그인 로그 엔티티
 */
@Entity
@Table(name = "member_login_log", indexes = [Index(name = "IDX_MEMBER_LOGIN_LOG_REQUEST_DATE", columnList = "request_date")])
@NamedQuery(name = "MemberLoginLog.findAll", query = "SELECT m FROM MemberLoginLog m")
class MemberLoginLog(
        @Column(nullable = false, length = 15)
        var ipv4: String,

        @Temporal(TemporalType.TIMESTAMP)
        @Column(name = "request_date", nullable = false)
        var requestDate: Date,

        @Column(nullable = false)
        var status: AuthStatusRule,

        @ManyToOne(fetch = FetchType.LAZY)
        @JoinColumn(name = "member_id", nullable = false)
        var member: Member
) : Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "log_id", unique = true, nullable = false)
    var logId: Long = 0

    companion object {
        private const val serialVersionUID = 1L
    }
}